### PR TITLE
Improve the logging config actor

### PR DESCRIPTION
**Pull request by llucax**
- **Make all arguments keyword-only**
- **Add a `name` argument to `LoggingConfigUpdatingActor`**
- **Sort arguments in `LoggingConfigUpdatingActor` constructor**
- **Initialize parent class after all attributes were initialized**
- **Add missing docstring for `LogLevel`**
- **Remove the `LoggingConfig.load` method**
- **Use relative imports**
- **Use the `dataclass` decorator from `dataclasses`**
- **Group config changes in the release notes**
- **Sort `__all__`**
- **Remove unnecessary required from metadata**

### CODE REVIEW

**Comment by llucax** [comment]
Ready for review, but based on #1126.

**Comment by llucax** [comment]
Also in preparation for a major revamp of the `config` module.

**Comment by llucax** [comment]
Ready for review and merging.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -11,7 +11,12 @@
     * Renamed to `LoggingConfigUpdatingActor` to follow the actor naming convention.
     * Make all arguments to the constructor keyword-only.
 
-- The `LoggingConfig.load` method was removed. Please use `frequenz.sdk.config.load_config()` instead.
+- `LoggingConfig`
+
+    * The `load()` method was removed. Please use `frequenz.sdk.config.load_config()` instead.
+    * The class is now a standard `dataclass` instead of a `marshmallow_dataclass`.
+
+- `LoggerConfig` is not a standard `dataclass` instead of a `marshmallow_dataclass`.
```
Comment (ela-kotulska-frequenz) [inline RELEASE_NOTES.md]: - `LoggerConfig` is **now** a standard `dataclass` instead of a `marshmallow_dataclass`.

**Review by ela-kotulska-frequenz — COMMENTED** [review]
LGTM just one comment in README.

**Review by ela-kotulska-frequenz — APPROVED** [review]

File: RELEASE_NOTES.md (hunk 1 above)
Comment (ela-kotulska-frequenz) [reply RELEASE_NOTES.md]: Can be fixed in next PR

**Comment by llucax** [comment]
Oh, I was just fixing it :D

**Pull request merged**